### ISSUE TITLE
File urls are not resolved as expected for a[href] attributes

### ISSUE THREAD

**Issue by medikoo** [tmpvar/jsdom#1141]
e.g. `foo.pdf` is resolved as `file://foo.pdf/` while it should be `file://foo.pdf`.

It's caused by this change -> https://github.com/tmpvar/jsdom/commit/ea8f5bf0b5d29fa32376e14f2e7dd6fb4b02e63a#diff-b4d2b12fce9c87d002a9a282c86134aeR175
as:
```javascript
url.resolve('file://foo.pdf', ''); // file://foo.pdf/
url.resolve('file://foo.pdf', '#'); // file://foo.pdf/#
```

**Comment by Sebmaster** [comment]
We're currently working on a new URL resolver (at jsdom/whatwg-url) anyways, which should fix all URL issues if they're handled by the URL spec.

**Comment by domenic** [comment]
It's a good test case to make sure we include though. Friggin file URLs...

**Comment by domenic** [comment]
Just added a test confirming this is fixed. Might have been fixed in a previous release in the 6.x series, but if nothing else it will be fixed in 6.3.0 coming out shortly.
